library/alloc/src/macros.rs

```diff
@@ -34,7 +34,7 @@
 /// be mindful of side effects.
 ///
 /// [`Vec`]: crate::vec::Vec
-#[cfg(not(test))]
+#[cfg(all(not(no_global_oom_handling), not(test)))]
```

Comment:
Small nit: I might prefer not(any(...)) here, which I think reads a little nicer?

I generally find it faster to digest or'd lists myself, but it probably varies from person to person, so feel free to resolve if you disagree.

Reply:
I used this one since `not(no_global_oom_handling)` is the one normally found elsewhere on its own (since it is used to disable things) and also makes it a prefix of the other (so perhaps it is easier to see it is the same thing but for `test`/`!test`):

```rust
#[cfg(all(not(no_global_oom_handling), not(test)))]
#[cfg(all(not(no_global_oom_handling), test))]
```

From a quick look, there are currently 2 `not(any(...))`, and 4 `all(not(...))` (for `no_global_oom_handling`/`test`).

I am fine with either, so please let me know which one you prefer given the above.

(Personally, I think multiple `cfg`s would be more readable and easier for diff/VCS purposes; and I hope for `&&`/`and` operators, but... :)

Reply:
Hm, yeah, not sure. Let's leave it as-is for now -- I can see either direction being beneficial. Multiple cfgs would also be fine I think (they don't work when you want an `or` but for `and` I think should work fine).